### DISCUSSION TITLE
How to add a label to a post?

### DISCUSSION THREAD

**Discussion by ronaldpk** [bluesky-social/atproto#1751] [How to do it?] [answered]
I'm posting a message with an image to bsky.app through the API. Is it possible to add a label like "nudity", "violence", "bad joke", to the post or to the image? If so, what labels are available, and what should the JSON for a labelled record look like? Browsing the lexicons got me to this (obviously wrong, just to show what I'm looking for):

```javascript
{
    "text": "a message here",
    "createdAt": "2023-10-15T16:32:34+09:00",
    "$type": "app.bsky.feed.post",
    "embed": {
        "$type": "app.bsky.embed.images",
        "images": [ { "alt": "alt text here" , "image": "uploaded image blob here" } ]
    },
    "labels": {
        "$type": "com.atproto.label.defs#selfLabels????",
        "values????" : [
            {
                "val????" : "possibly offending???"
            }
        ]
    }
}
```

**Comment by bnewbold** [comment, marked as answer]
The currently-implemented label values are documented here:

https://github.com/bluesky-social/atproto/blob/main/packages/api/docs/labels.md

We don't yet have a collaborative development or governance process for extending this list with additional options, but expect to have a process in place eventually. On the one hand it would be good to keep the vocabulary flexible to diverse needs, on the other hand having some subject experts involved and rough consensus on popular values would be good for ecosystem consistency. Content moderation has a rich history to adapt and learn from.

You can find some pseudo-code examples of what the objects look like by searching this repo (note thee are in TypeScript syntax, not JSON directly). Eg:

https://github.com/bluesky-social/atproto/blob/2af3f6c783d18dd49c9ae8610cc0738b202fb672/packages/bsky/tests/views/__snapshots__/posts.test.ts.snap#L51

**Reply by ronaldpk** [reply]
Thanks for the links. I searched the repo but overlooked that documentation page.
